ampower_farmaans/ampower_farmaans/doctype/template_docx/template_docx.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2023, Ambibuzz Technologies LLP and contributors
+# For license information, please see license.txt
+import frappe
+from frappe.model.document import Document
+from frappe.utils.file_manager import save_file
+import ampower_farmaans
+import json
+from docxtpl import DocxTemplate
+import io
+
+class TemplateDocx(Document):
+    pass
+
+@frappe.whitelist()
+def attach_document(template_ref, cur_document):
+    context = json.loads(cur_document)
+    template_file = frappe.get_doc("File", {"file_url": template_ref})
+    filename = template_file.get_full_path()
+    # creates a object using DocxTemplate class
+    docm = DocxTemplate(filename)
+    # puting the data as a dictionary for jinja variables
+    docm.render(context)
+    # to write and save .docx file in frappe the data should be byte
+    with io.BytesIO() as output_buffer:
+        docm.save(output_buffer)
+        content = output_buffer.getvalue()
+        prepare_file = save_file(
+                        f'{context["name"]}.docx', 
+                        content=content,
+                        dt=str(context["doctype"]),
+                        dn=str(context["name"]))
+    return {
+        "prepare_file": prepare_file
+       }
+
+@frappe.whitelist()
+def download_document(template_ref, cur_document):
```

Comment:
the attach function is used for attaching generated print file, if we use the same code then user experience will not good, because user click for download and our function do both attaching and downloading, so for that I used separate functions.